pennylane/qchem/tapering.py

```diff
@@ -538,3 +539,113 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
+
+    >>> symbols, geometry = ['He', 'H'], np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4589]])
+    >>> mol = qchem.Molecule(symbols, geometry, charge=1)
+    >>> H, n_qubits = qchem.molecular_hamiltonian(symbols, geometry)
+    >>> generators = qchem.symmetry_generators(H)
+    >>> paulixops = qchem.paulix_ops(generators, n_qubits)
+    >>> paulix_sector = qchem.optimal_sector(H, genera  tors, mol.n_electrons)
+    >>> qchem.taper_operation(qml.SingleExcitation(1, wires=[0, 2]),
+                                generators, paulixops, paulix_sector, wire_order=H.wires)
+    [PauliRot(0.5+0.j, 'RY', wires=[0])]
+
+    This can even be used for within a :class:`~.pennylane.QNode`:
+
+    >>> dev = qml.device('default.qubit', wires=[0, 1])
+    >>> @qml.qnode(dev)
+    ... def circuit(params):
+    ...     qchem.taper_operation(qml.DoubleExcitation(params[0], wires=[0, 1, 2, 3]),
+    ...                             generators, paulixops, paulix_sector, H.wires)
+    ...     return qml.expval(qml.PauliZ(0)@qml.PauliZ(1))
+    >>> drawer = qml.draw(circuit, show_all_wires=True)
+    >>> print(drawer(params=[0.38686753]))
+        0: ─╭RXY(-0.10+0.00j)─╭RYX(-0.10+0.00j)─┤ ╭<Z@Z>
+        1: ─╰RXY(-0.10+0.00j)─╰RYX(-0.10+0.00j)─┤ ╰<Z@Z>
+    """
+
+    if gen_op is None:
+        if operation.num_params < 1:  # Non-parameterized gates
+            gen_mat = -1j * scipy.linalg.logm(qml.matrix(operation, wire_order=wire_order))
+            gen_op = qml.Hamiltonian(
+                *qml.utils.decompose_hamiltonian(gen_mat, wire_order=wire_order, hide_identity=True)
+            )
+            qml.simplify(gen_op)
+            if gen_op.ops[0].label() == qml.Identity(wires=[wire_order[0]]).label():
+                gen_op -= qml.Hamiltonian([gen_op.coeffs[0]], [qml.Identity(wires=wire_order[0])])
+        else:  # Single-parameter gates
+            try:
+                gen_op = qml.generator(operation, "hamiltonian")
+            except ValueError as exc:
+                raise NotImplementedError(
+                    f"Generator for {operation} is not implemented, please provide it with 'gen_op' args."
+                ) from exc
+    else:
+        if not isinstance(gen_op, qml.Hamiltonian):
+            raise ValueError(
+                f"Generator for the operation needs to a qml.Hamiltonian, but got {type(gen_op)}."
+            )
+        mat1 = scipy.linalg.expm(1j * qml.matrix(gen_op, wire_order=wire_order))
+        mat2 = qml.matrix(operation, wire_order=wire_order)
+        phase = np.divide(mat1, mat2, out=np.zeros_like(mat1, dtype=complex), where=mat1 != 0)[
+            np.nonzero(mat1)
+        ]
+        if not np.allclose(phase / phase[0], np.ones(len(phase))):  # check if the phase is global
+            raise ValueError(
+                f"Given gen_op: {gen_op} doesn't seem to be the correct generator for the {operation}."
+            )
+
+    if np.all(
+        [
+            [
+                qml.grouping.is_commuting(op1, op2)
```

Comment:
```suggestion
                qml.is_commuting(op1, op2)
```
(once latest master is merged here now that #3033 was merged)